### PR TITLE
chore: update registry1 weekly to 0.13.1, API, and testing workflows

### PR DESCRIPTION
**Pull request by justinthelaw**
## Description

Updates everything Registry1 to v0.13.0, and fixes the Registry1 weekly to pin all components to 0.13.1 (or whatever the LEAPFROGAI_VERSION is) so proper testing can be performed. Removed integration testing, since it uses a local Python instance of the API, which is out-of-scope. Adds a `zarf-config.yaml` for non-upstream flavor builds to centralize un-mutable versioning.

Fixes `release-please-config.json` to also update DEVELOPMENT.md, where it references the LeapfrogAI latest tag as part of an example code block. Adds Chainguard login so we are not rate-limited, and adds granular permissions blocks for the associated workflows. Centralizes model name in the `conftest.py` and generation configurations within the `test_llm_generation.py`. Adds better clean Makefile target for all cached test artifacts, and improves Makefile build pattern for `registry1` flavors.

### BREAKING CHANGES

- uses a `zarf-config.yaml` when building the API `registry1` flavor to centralize un-mutable version (only mutates via release-please)
  - makes it easier to do registry1 (uses the `v`) versus upstream flavor images by variabilizing the version number

### CHANGES

- bumps all registry1 deps to 0.13.1
- changes name of workflow to match pattern of `[frequency]-[purpose].yaml`
- fixes pinning and testing technique of the Registry1 image(s)
- remove out-of-scope integration testing from the Registry1 weekly
- fixes release-please config to also update DEVELOPMENT.md code block
- adds chainguard login to the UDS setup tasks
- modifies and uses the `utils.client.py` for all e2e tests
- centralizes generation params for the tests within `test_llm_generation.py`
- adds better clean Makefile target for all cached test artifacts
- improves Makefile build pattern for `registry1` flavor
- removes the `udsCliVersion` and instead allows the default UDS CLI version packaged with the UDS Common action, allowing renovate to update that automatically, instead of us doing it manually
- adds env `LEAPFROGAI_MODEL: llama-cpp-python` to e2e tests that need a specific model identified

## Related Issue

Relates to https://repo1.dso.mil/dsop/opensource/defenseunicorns/leapfrogai/api/-/issues/22

## Checklist before merging

- [X] Tests, documentation, ADR added or updated as needed
- [X] Followed the [Contributor Guide Steps](https://github.com/defenseunicorns/leapfrogai/blob/main/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by justinthelaw** [comment]
Still being blocked by the following, unmerged PR and issue: https://repo1.dso.mil/dsop/opensource/defenseunicorns/leapfrogai/api/-/issues/23

Waiting on the IronBank team to review and merge before re-running the Registry1 Weekly tests.

**Comment by justinthelaw** [comment]
Blocked by https://repo1.dso.mil/dsop/opensource/defenseunicorns/leapfrogai/api/-/merge_requests/39

File: tests/e2e/conftest.py [hunk 1]
```diff
@@ -1,12 +1,32 @@
...
+
+def get_model_name():
+    model_name = os.getenv("LEAPFROGAI_MODEL")
+    if model_name is None:
+        warnings.warn(
+            f"LEAPFROGAI_MODEL environment variable not set. Defaulting to '{DEFAULT_LEAPFROGAI_MODEL}'.\n"
+            "Consider setting LEAPFROGAI_MODEL explicitly. Examples: 'vllm', 'repeater', 'llama-cpp-python'."
+        )
+        model_name = DEFAULT_LEAPFROGAI_MODEL
+    return model_name
```
Comment (gphorvath) [inline tests/e2e/conftest.py:27]: I actually have a client util to do something like this if you want to wait for my PR to go in: 

https://github.com/defenseunicorns/leapfrogai/blob/1b8040a97505f2eee67338fe6947e1367e5fa363/tests/utils/client.py

Comment (justinthelaw) [reply tests/e2e/conftest.py:27]: Yes, please merge yours in first!

**Review by CollectiveUnicorn — APPROVED** [review]

File: tests/e2e/conftest.py (hunk 1 above)
Comment (gphorvath) [reply tests/e2e/conftest.py:27]: it should be merged, not sure if you want to cut over

**Review by gphorvath — DISMISSED** [review]

**Review by gphorvath — DISMISSED** [review]

**Review by CollectiveUnicorn — DISMISSED** [review]

**Review by CollectiveUnicorn — APPROVED** [review]

File: tests/utils/client.py [hunk 2]
```diff
@@ -1,9 +1,113 @@
...
+    Raises:
+        AssertionError: If the request fails or the response status code is not 200.
+    """
+
+    url = f"{supabase_base_url}/auth/v1/token?grant_type=password"
+    headers = {"apikey": f"{api_key}", "Content-Type": "application/json"}
+    data = {"email": test_email, "password": test_password}
+
+    response = requests.post(url, headers=headers, json=data)
+    if response.status_code != 200:
```
Comment (gphorvath) [inline tests/utils/client.py:103]: Suggestion:
```
from fastapi import status

if response.status_code != status.HTTP_200_OK:
```

File: tests/utils/client.py [hunk 3]
```diff
@@ -12,7 +117,15 @@ def get_leapfrogai_model() -> str:
         str: The model to use for LeapfrogAI. (default: "vllm")
     """
 
-    return os.getenv("LEAPFROGAI_MODEL", "vllm")
+    model = os.getenv("LEAPFROGAI_MODEL")
+
+    if model is None:
```
Comment (gphorvath) [inline tests/utils/client.py:122]:
```suggestion
    if not model:
```

File: tests/utils/client.py [hunk 4]
```diff
@@ -49,14 +162,18 @@ def get_leapfrogai_api_key() -> str:
...
     Returns:
         str: The API key for the LeapfrogAI API.
+
     Raises:
         ValueError: If LEAPFROGAI_API_KEY or SUPABASE_USER_JWT is not set.
     """
 
     api_key = os.getenv("LEAPFROGAI_API_KEY") or os.getenv("SUPABASE_USER_JWT")
 
     if api_key is None:
```
Comment (gphorvath) [inline tests/utils/client.py:172]:
```suggestion
    if not api_key:
```

**Review by gphorvath — APPROVED** [review]

**Pull request merged**